### PR TITLE
Fixes actor ref leak in AddressTerminatedTopic (#4304)

### PR DESCRIPTION
**Pull request by petrikero**
I made the PR against v1.3 as that's what we're using and where I could confirm the bug and the fix.

### CODE REVIEW

**Comment by petrikero** [comment]
The failing .NET Framework Unit Tests seem to me to be unrelated, but I'm not sure I'm reading the outputs correctly.

**Comment by Aaronontheweb** [comment]
@petrikero the v1.3 branch is missing a bunch of racy spec fixes we introduced in the v1.4 / dev branch. I'll take a look at these.

**Comment by petrikero** [comment]
Thanks! The issues are crashing our systems after a few days of running, so they're pretty critical for us. I did manage to inject a custom-built Akka.dll (with the two PRs included) into our build pipeline to solve the bleeding for now.

If there's anything more I can do to help get them merged, please let me know. I'll be happy to help, but am a bit at a loss with the size of the project. :)

**Comment by Aaronontheweb** [comment]
failed specs appear to be largely the result of:
- Multiple bind on port errors (fixed in v1.4 branch)
- Specs that are tightly temporally coupled - most of them have been disabled in v1.4 branch.

File: src/core/Akka/Actor/ActorCell.DeathWatch.cs [hunk 1]
```diff
@@ -173,7 +173,10 @@ protected void TellWatchersWeDied()
             }
             finally
             {
-                _state = _state.ClearWatching();
+                MaintainAddressTerminatedSubscription(() =>
```
Comment (Aaronontheweb) [inline src/core/Akka/Actor/ActorCell.DeathWatch.cs]: Wow, this is an old bug. Looks right though.

**Comment by Aaronontheweb** [comment]
I'll see about having us to a v1.3.18 release to fix these.

**Pull request merged**